server/models/submissions/getContestLeaderboard.js

```diff
@@ -0,0 +1,117 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {Array} filterParams
+ * @return {String}
+ */
+function filterQuery(filterParams) {
+  if (!Array.isArray(filterParams)) {
+    return ''
+  }
+  return filterParams
+    .map(({ id, value }) =>
+      value.length
+        ? ` AND ${id} IN (${value
+            .map((val) => (typeof val === 'string' ? `'${val}'` : val))
+            .join(',')}) `
+        : ''
+    )
+    .join('')
+}
+
+/**
+ *
+ * @param {Array} sortParams
+ * @return {String}
+ */
+function sortQuery(sortParams) {
+  if (!Array.isArray(sortParams)) {
+    return ''
+  }
+  return (
+    ' ORDER BY ' +
+    sortParams
+      .map(
+        ({ id, desc }) =>
+          `${id === 'rank' ? '`rank`' : id} ${desc ? 'DESC' : 'ASC'}`
+      )
+      .join(', ')
+  )
+}
```

Comment:
Seems like the `filterQuery` and `sortQuery` functions are used in many places...and almost same too...better move it to utils:
`/server/models/utils`
This will reduce code duplicacy.

Reply:
Actually, I thought of that, but the problem is that there are some minor modifications for every end point. 
Like we need to treat `rank` keyword specially for this query. 
For the query in `server/models/submissions/getAllSubmissions.js`, there we need to treat another column name separately in sortQuery, and use `HAVING` for that column, as it was an alias name there.
Keeping this way would make faster

Reply:
ok